src/twisted/words/test/test_irc.py

```diff
@@ -2572,6 +2572,67 @@ def test_dccChatIndecipherablePort(self):
         self.assertEqual(str(result), "Indecipherable port 'sd@d'")
 
 
+UNICODE_REPLACEMENT_CHARACTER = "\ufffd"
+
+
+class ClientDecodingTests(IRCTestCase):
+    """
+    Tests for L{IRCClient} receiving data in different encodings.
+    """
+
+    def setUp(self):
+        methods = ["handleCommand", "badMessage"]
+        self.client = CollectorClient(methods)
+
+    def testDecodingUnspecifiedCodec(self):
+        """
+        Message decoding doesn't fail when no fitting codec was specified and
+        default decode error handling is used. Characters that can not be
+        decoded with UTF8 will be replaced by the unicode replacement
+        character 'U+FFFD'.
+        """
+        self.client.decodeCodecs = IRCClient.decodeCodecs
+        self.client.decodeFallbackErrorhandling = IRCClient.decodeFallbackErrorhandling
+        prefix = ":foo!bar@baz"
+        line_bytes = f"{prefix} PRIVMSG foo :ä".encode("latin-1")
+        self.client.lineReceived(line_bytes)
+        self.assertEqual(
+            self.client.methods,
+            [
+                (
+                    "handleCommand",
+                    ("PRIVMSG", prefix[1:], ["foo", UNICODE_REPLACEMENT_CHARACTER]),
+                )
+            ],
+        )
+
+    def testDecodingSpecifiedCodec(self):
+        """
+        Message decoding will try the specified codecs before using
+        the fallback errorhandling.
+        """
+        self.client.decodeCodecs = ["utf-8", "latin-1"]
+        self.client.decodeFallbackErrorhandling = IRCClient.decodeFallbackErrorhandling
+        prefix = ":foo!bar@baz"
+        line_bytes = f"{prefix} PRIVMSG foo :ä".encode("latin-1")
+        self.client.lineReceived(line_bytes)
```

Comment:
So right now our standard requires ASCII... I filed #11723 to change this. Also the variable should be named `lineBytes`.

The `\N{}` syntax can be handy here, though in this case it might be best to use a `b"\xe4"` literal as it's more obvious that this isn't valid UTF-8.

```suggestion
        lineBytes = f"{prefix} PRIVMSG foo :\N{LATIN SMALL LETTER A WITH DIAERESIS}".encode("latin-1")
        self.client.lineReceived(lineBytes)
```

The same goes for the other tests.

Reply:
I didn't know about the `\N{}` syntax. You're right about using the bytes literal directly being more obvious, so I changed these test that way.